api/samples/bbbuf.c

```diff
@@ -67,6 +67,8 @@ event_app_instruction(void *drcontext, void *tag, instrlist_t *bb, instr_t *inst
     /* We need a 2nd scratch reg for several operations on AArch32 and AArch64 only. */
     reg_id_t reg2 = DR_REG_NULL;
 
+    drmgr_disable_auto_predication(drcontext, bb);
```

Comment:
Sure, if we see that the feature is too confusing or we see that in the future bugs start arising because new clients should have opted out yet forget to do so, we should make this feature opt-in.